app/models/pipeline_output.rb

```diff
@@ -43,4 +48,18 @@ def generate_aggregate_counts(tax_level_name)
       GROUP BY 1,2,3,4,5"
     )
   end
+
+  def download_taxid_fasta(taxid)
+    # currently only works with species-level taxids (ANNOTATED_FASTA is species-level)
+    input_fasta_s3_path = "#{sample.sample_output_s3_path}/#{ANNOTATED_FASTA}"
+    local_fasta_path = "#{LOCAL_FASTA_PATH}/#{id}/#{taxid}"
+    local_input = "#{local_fasta_path}/#{ANNOTATED_FASTA}"
+    local_output = "#{local_fasta_path}/#{TAXID_FASTA}"
+    command = "mkdir -p #{local_fasta_path};"
+    command += "aws s3 cp #{input_fasta_s3_path} #{local_fasta_path}/;"
+    command += "grep -A 1 -E 'nr:#{taxid}:|nt:#{taxid}:' #{local_input} | sed '/^--$/d' > #{local_output}"
```

Comment:
FYI the data there is not taxon id, it's accession id actually.

Reply:
yep, I'm making a new file in the pipeline